src/librustc_infer/infer/lexical_region_resolve/mod.rs

```diff
@@ -874,15 +877,22 @@ impl<'cx, 'tcx> LexicalResolver<'cx, 'tcx> {
         // If we have a scenario like `exists<'a> { forall<'b> { 'b:
         // 'a } }`, we wind up without any lower-bound -- all we have
         // are placeholders as upper bounds, but the universe of the
-        // variable `'a` doesn't permit those placeholders.
+        // variable `'a`, or some variable that `'a` has to outlive, doesn't
+        // permit those placeholders.
+        let min_universe = lower_vid_bounds
+            .into_iter()
+            .map(|vid| self.var_infos[vid].universe)
+            .min()
+            .expect("lower_vid_bounds should at least include `node_idx`");
```

Comment:
This is I guess similar to the change I made in the NLL checker, where we require that if `'a: 'b` then `universe('a) <= universe('b)` (or vice versa, I can't remember). Seems fine, but given that we have an explicit representation of empty in a given universe, less necessary here, right?

Reply:
The problem is that the `collect_bounding_regions` doesn't return the `ReEmpty` lower bounds because they're not represented in the `RegionGraph`.